packages/server/src/browser-manager.ts

```diff
@@ -0,0 +1,37 @@
+import { Operation, resource } from 'effection';
+import { Deferred } from '@bigtest/effection';
+import { Local, Options as WebDriverOptions, WebDriver } from '@bigtest/webdriver';
+
+interface CreateOptions {
+  connectURL: string;
+  drivers: Record<string, WebDriverOptions>;
+  launch: Record<string, Partial<WebDriverOptions>>;
+}
+
+export interface BrowserManager {
+  ready(): Operation<void>;
+}
+
+export function* createBrowserManager(options: CreateOptions): Operation<BrowserManager> {
+
+  let ready = Deferred<void>();
+
+  let manager: BrowserManager = {
+    *ready() { yield ready.promise; }
+  }
+
+  return yield resource(manager, function*() {
+    let launches = Object.keys(options.launch)
+      .map(key => Object.assign({}, options.drivers[key], options.launch[key]) as WebDriverOptions);
+
+    for (let launch of launches) {
+      let driver: WebDriver = yield Local(launch);
+      yield driver.navigateTo(options.connectURL);
+    }
+
+    ready.resolve();
+
+    yield;
+  })
```

Comment:
This is a bit of a departure from the current design of the orchestrator, where it doesn't use resources, but rather the `delegate` pattern to report readiness. I think that our experience with resources shows that they are nicer to work with and easier to understand, so this is probably a good thing, still wanted to call it out so we can think about if this is the direction we want to evolve the orchestrator into as well.

Reply:
I agree 100%. I think symmetry of code is often more important than using the latest and greatest. That said, using resources and here felt _amazing_ here. I feel very strongly that this should be the new normal for our other services. As I see it, our options are:

1) go with this, and then refactor the other "raw forks" to become resources later.
2) refactor other "raw forks" into resources now, then rebase this onto that.
3) re-implement this using the "raw fork" model and then refactor all of them together in one go.